subprojects/distributions/src/integTest/groovy/org/gradle/BinaryBreakageIntegrationTest.groovy

```diff
@@ -0,0 +1,133 @@
+import japicmp.model.JApiClass
+import japicmp.output.stdout.StdoutOutputGenerator
+import org.gradle.api.internal.file.pattern.PathMatcher
+import org.gradle.api.internal.file.pattern.PatternMatcherFactory
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import org.gradle.integtests.fixtures.executer.GradleDistribution
+import org.gradle.integtests.fixtures.executer.UnderDevelopmentGradleDistribution
+import org.gradle.integtests.fixtures.versions.ReleasedVersionDistributions
+import spock.lang.Unroll
+
+import static org.junit.Assume.assumeTrue
+
+@Unroll
+class BinaryBreakageIntegrationTest extends AbstractIntegrationSpec{
+
+    private static final List<String> PUBLIC_API_INCLUDES = System.getProperty('org.gradle.public.api.includes').split(':')
+    private static final List<String> PUBLIC_API_EXCLUDES = System.getProperty('org.gradle.public.api.excludes').split(':')
+    private static final Map<String, String> ACCEPTED_CHANGES = [
+        'org.gradle.api.initialization.ConfigurableIncludedBuild': '''---! REMOVED INTERFACE: PUBLIC(-) ABSTRACT(-) org.gradle.api.initialization.ConfigurableIncludedBuild  (not serializable)
+\t---! REMOVED INTERFACE: org.gradle.api.initialization.IncludedBuild
+\t---! REMOVED METHOD: PUBLIC(-) ABSTRACT(-) void dependencySubstitution(org.gradle.api.Action)''',
+
+        'org.gradle.api.initialization.IncludedBuild': '''---! REMOVED INTERFACE: PUBLIC(-) ABSTRACT(-) org.gradle.api.initialization.IncludedBuild  (not serializable)
+\t---! REMOVED METHOD: PUBLIC(-) ABSTRACT(-) java.lang.String getName()
+\t---! REMOVED METHOD: PUBLIC(-) ABSTRACT(-) java.io.File getProjectDir()
+\t---! REMOVED METHOD: PUBLIC(-) ABSTRACT(-) org.gradle.api.tasks.TaskReference task(java.lang.String)''',
+
+        'org.gradle.api.invocation.Gradle': '''***! MODIFIED INTERFACE: PUBLIC ABSTRACT org.gradle.api.invocation.Gradle  (not serializable)
+\t***! MODIFIED METHOD: PUBLIC ABSTRACT org.gradle.includedbuild.IncludedBuild (<-org.gradle.api.initialization.IncludedBuild) includedBuild(java.lang.String)''',
+        'org.gradle.StartParameter': '''***! MODIFIED CLASS: PUBLIC org.gradle.StartParameter  (default serialVersionUID changed)
+\t---! REMOVED INTERFACE: org.gradle.initialization.ParallelismConfiguration'''
+    ]
+
+    private List<PathMatcher> publicApiIncludeMatchers
+
+    private List<PathMatcher> publicApiExcludeMatchers
+
+    def setup() {
+        publicApiIncludeMatchers = PUBLIC_API_INCLUDES.collect { PatternMatcherFactory.compile(true, it) }
+        publicApiExcludeMatchers = PUBLIC_API_EXCLUDES.collect { PatternMatcherFactory.compile(true, it) }
+        assert !publicApiIncludeMatchers.empty
+        assert !publicApiExcludeMatchers.empty
+    }
+
+    def "should be binary compatibility with Gradle #version"() {
+        GradleDistribution current = new UnderDevelopmentGradleDistribution()
+
+        // For a hotfix release we need to trust our own judgement or run this test manually
```

Comment:
⭕️ This comment doesn't really help me understand the line below, I'd just remove it.